components/board/fake/board.go

```diff
@@ -235,7 +235,7 @@ func (b *Board) StreamTicks(ctx context.Context, interrupts []board.DigitalInter
 	extra map[string]interface{},
 ) error {
 	for _, i := range interrupts {
-		i.AddCallback(ch)
+		AddCallback(i.(*DigitalInterruptWrapper), ch)
```

Comment:
If `i` isn't actually a `DigitalInterruptWrapper`, this is going to panic. and it kinda looks like any `board.DigitalInterrupt` could be passed in here, including injected ones during a test.

Reply:
If injected digital interrupts are being used in a test why would someone want to use a fake board instead of an injected board? But yes a user could potentially call this and use a different type of interrupt.